### PR TITLE
refactor(lsp): share invalid utf8 exceptions

### PR DESCRIPTION
**Pull request by rgrinberg**

### CODE REVIEW

File: ocaml-lsp-server/src/document.ml [hunk 1]
```diff
@@ -222,14 +222,15 @@ let make wheel config pipeline (doc : DidOpenTextDocumentParams.t)
...
-  | exception Text_document.Invalid_utf8 ->
+  | exception Text_document.Invalid_utf message ->
     Log.log ~section:"warning" (fun () ->
         Log.msg
           "dropping update due to invalid utf8"
           [ ( "changes"
             , Json.yojson_of_list
                 TextDocumentContentChangeEvent.yojson_of_t
                 changes )
+          ; ("message", `String message)
```
Comment (ulugbekna) [inline ocaml-lsp-server/src/document.ml:233]: I wonder whether this log message could be a bit more helpful if we reported the context around the exception - on what document, after/on which changeEvent and with what doc state the error occurred

File: lsp/src/text_document.ml [hunk 2]
```diff
@@ -32,8 +32,9 @@ let find_utf8_pos =
     if char = 0 then Uutf.decoder_byte_count dec
     else
       match Uutf.decode dec with
-      | `Malformed _ | `Await -> raise Invalid_utf8
-      | `End -> assert false
+      | `Malformed m -> raise (Invalid_utf m)
```
Comment (ulugbekna) [inline lsp/src/text_document.ml:35]: When `Uutf.decode` returns `Malformed s`, what's `s` here? Just bytes read with `Uutf.Manual.src` call?

**Review by ulugbekna — APPROVED** [review]

File: lsp/src/text_document.ml (hunk 2 above)
Comment (rgrinberg) [reply lsp/src/text_document.ml:35]: The malformed byte sequence.

File: lsp/src/text_document.ml [hunk 3]
```diff
@@ -32,8 +32,9 @@ let find_utf8_pos =
...
+      | `Await -> raise (Invalid_utf "not enough input")
+      | `End -> raise (Invalid_utf "not enough input")
```
[the remark this answers is not in the document]
Comment (rgrinberg) [reply lsp/src/text_document.ml]: I'm not even sure how to trigger these errors at the moment, so I hate to spend time improving them until I'm certain they have some use.

**Comment by rgrinberg** [comment]
I improved the messages somewhat. Feel free to improve things further.

**Pull request merged**